drivers/hmis/spec/requests/hmis/submit_form_spec.rb

```diff
@@ -555,6 +555,36 @@ def submit_enrollment_form(input)
     end
   end
 
+  describe 'SubmitForm for Enrollment on project with ProjectAutoEnterConfig' do
+    let(:definition) { Hmis::Form::Definition.find_by(role: :ENROLLMENT) }
+    let!(:aec) { create :hmis_project_auto_enter_config, project: p1 }
+    let(:test_input) do
+      {
+        form_definition_id: definition.id,
+        **completed_form_values_for_role(:ENROLLMENT),
+        project_id: p1.id,
+        client_id: c1.id,
+        confirmed: false,
+      }
+    end
+
+    it 'should save new enrollment without WIP status' do
+      response, result = post_graphql(input: { input: test_input }) { mutation }
+      errors = result.dig('data', 'submitForm', 'errors')
+      expect(response.status).to eq(200), result.inspect
+      expect(errors).to be_empty
+      record = result.dig('data', 'submitForm', 'record')
+      expect(record['inProgress']).to eq(false)
+      e_id = Hmis::Hud::Enrollment.find(record['id']).enrollment_id
```

Comment:
This is such an FAQ and I have been meaning to write it up. Finally just did here https://github.com/greenriver/hmis-warehouse/pull/4047 (and https://github.com/greenriver/hmis-warehouse/pull/4048!) let me know if you have further questions.

We [resolve](https://github.com/greenriver/hmis-warehouse/blob/1c194524832860549913fc8f9194115ae3d001c2/drivers/hmis/app/graphql/types/hmis_schema/enrollment.rb#L78) the database id (Enrollment.id) as the record id on the graphql Enrollment type. We never show the "HUD Key" for Enrollment in the HMIS application, so we don't even resolve it. We do use the ProjectID and OrganizationID in the interface in a few places, because it's helpful for migrated-in data to have those IDs match the old system.

Reply:
Ok, I think that makes sense -- thank you for the write-up, it's really helpful!
Does it make sense then, that in order to resolve the `CustomAssessment` that's associated with this enrollment, I've had to 'translate' from the database's ID to the hmis's UUID?

Reply:
Yep it does. I think it could be adjusted to be a bit more readable, though:

```
      record = result.dig('data', 'submitForm', 'record')
      expect(record['inProgress']).to eq(false)
      e_id = Hmis::Hud::Enrollment.find(record['id']).enrollment_id
      ca = Hmis::Hud::CustomAssessment.where(enrollment_id: e_id).last
      expect(ca).not_to be_nil
      expect(ca.wip).to eq(false)
      fp = Hmis::Form::FormProcessor.where(custom_assessment: ca).last
      expect(fp).not_to be_nil
```
=>
```
      enrollment_id = result.dig('data', 'submitForm', 'record', 'id')

      enrollment = Hmis::Hud::Enrollment.find_by(id: enrollment_id)
      expect(enrollment).to be_present
      expect(enrollment.in_progress?).to eq(false)
      expect(enrollment.intake_assessment).to be_present
      expect(enrollment.intake_assessment.assessment_date).to eq(enrollment.entry_date)
      expect(enrollment.intake_assessment.wip).to eq(false)
      expect(enrollment.intake_assessment.form_processor).to be_present
```

Reply:
Oh great, this is much better!